omnimatter_compression/function.lua

```diff
@@ -89,7 +89,8 @@ local function find_result_icon(raw_item)
     elseif raw_item.icon then
       return {{
         icon = raw_item.icon,
-        icon_size = raw_item.icon_size
+        icon_size = raw_item.icon_size,
+        icon_mipmaps = raw_item.icon_mipmaps
```

Comment:
You'll want to add a check if mipmaps exist since this will probably crash on items that define icon, but don't use mipmaps.

Reply:
Is that necessary? The code only gets this far if there is not an `icons` field, so we're just dealing with a standard flat icon at this point. Mipmaps is an optional field, and so the code as written just means that `icon_mipmaps` would be nil, since in this hypothetical, `raw_item.icon_mipmaps = nil`, so the return would be `icon` and `icon_size`. 

The game treats a nil `icon_mipmaps` field as 1. I don't think not checking for the existence of raw_item.icon_mipmaps would case a startup error.

Reply:
Since the raw_item.icon_mipmaps is a number, not a table, we don't risk trying to index a nil table. If it *is* nil, the game simply applies default behavior as if it hadn't been defined in the prototype (the correct outcome). Additionally, since this is the condition for `icon`, not `icons`, we don't risk setting a value that already exists to nil.

Reply:
Thanks for the clarification. I had no idea about this usage, as mostly the game seems to crash when you fail to remember a check somewhere. Looking good then :)